dpgen/ops/dispatcherop.py

```diff
@@ -0,0 +1,118 @@
+from abc import abstractmethod
+from pathlib import Path
+from typing import List, Set
+from itertools import chain
+
+from dpdispatcher import Machine, Resources, Task, Submission
+
+from .op import OP, Status
+from .opio import OPIO
+from .context import IterationContext
+
+
+class DispatcherOP(OP):
+    r"""This OP will call DPDispatcher to execuate commands on remote.
+    
+    Parameters
+    ----------
+    context : IterationContext
+        DP-GEN iteration context
+    machine : dict
+        DPDispatcher machine parameters. See also DPDispatcher doc.
+    resources : dict
+        DPDispatcher machine parameters. See also DPDispatcher doc.
+    command : str
+        path or command to a certain program on remote
+    """
+    @OP.set_status(status = Status.INITED)
+    def __init__(
+            self,
+            context: IterationContext,
+            machine: dict,
+            resources: dict,
+            command: str,
+    )->None:
+        super().__init__(context)
+        self.machine = Machine(**machine)
+        self.resources = Resources(**resources)
+        self.user_command = command
+    
+    @property
+    def command(self) -> str:
+        """str: The full bash command produced by the command the user provides."""
+        return self.user_command
+    
+    @property
+    def outlog(self) -> str:
+        return "log"
+
+    @property
+    def errlog(self) -> str:
+        return "log"
+    
+    @property
+    @abstractmethod
+    def forward_files(self) -> Set[Path]:
+        """List[str]: relative path to task path"""
+
+    @property
+    @abstractmethod
+    def backward_files(self) -> Set[Path]:
```

Comment:
abstractmethod does not have `self` as input

Reply:
Do you mean `staticmethod` does not have `self` as input?

Reply:
yes

Reply:
but this is `abstractmethod` and not `staticmethod`...